qfdmo/views/adresses.py

```diff
@@ -122,6 +122,8 @@ def get_context_data(self, **kwargs):
         # Manage the selection of sous_categorie_objet and actions
         acteurs = self._manage_sous_categorie_objet_and_actions()
 
+        acteurs = acteurs.prefetch_related("action_principale")
```

Comment:
Evite de faire des appels en DB pour chacun des acteurs affichés sur la carte

Reply:
Tu as profilé le nombre de requêtes ? 
Normalement si tu modifies la queryset derrière, genre avec un appel à `.filter`, ça annule les effets de `prefetch_related`. 

Je suis pas 100% sûr, mais ça se vérifie avec la django debug toolbar.

[Cf docs](https://docs.djangoproject.com/en/5.1/ref/models/querysets/#prefetch-related)

> Note
> 
> Remember that, as always with QuerySets, any subsequent chained methods which imply a different database query will ignore previously cached results, and retrieve data using a fresh database query. So, if you write the following:
> 
> `pizzas = Pizza.objects.prefetch_related("toppings")`
> `[list(pizza.toppings.filter(spicy=True)) for pizza in pizzas]`
> …then the fact that pizza.toppings.all() has been prefetched will not help you. The `prefetch_related('toppings')` implied `pizza.toppings.all()`, but `pizza.toppings.filter()` is a new and different query. The prefetched cache can’t help here; in fact it hurts performance, since you have done a database query that you haven’t used. So use this feature with caution!
> 
> Also, if you call the database-altering methods [add()](https://docs.djangoproject.com/en/5.1/ref/models/relations/#django.db.models.fields.related.RelatedManager.add), [create()](https://docs.djangoproject.com/en/5.1/ref/models/relations/#django.db.models.fields.related.RelatedManager.create), [remove()](https://docs.djangoproject.com/en/5.1/ref/models/relations/#django.db.models.fields.related.RelatedManager.remove), [clear()](https://docs.djangoproject.com/en/5.1/ref/models/relations/#django.db.models.fields.related.RelatedManager.clear) or [set()](https://docs.djangoproject.com/en/5.1/ref/models/relations/#django.db.models.fields.related.RelatedManager.set), on related managers, any prefetched cache for the relation will be cleared.

Reply:
👍 

Oui, j'ai vérifié qu'il n'y ai pas plus de requêtes avant et après avec debug tool bar sur une url qui affiche une carte

Je neppense pas que les prefetch_related qui ne sont pas concernés par le filtres soient annulé

Reply:
cool, ok pour moi